src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -1288,5 +1288,26 @@ public static bool All<TSource>(this ReadOnlySpan<TSource> source, Func<TSource,
             }
             return true;
         }
+
+        public static void CopyPropertiesTo<T, TU>(this T source, TU dest)
+        {
+            var sourceProps = typeof(T).GetProperties().Where(x => x.CanRead).ToList();
+            var destProps = typeof(TU).GetProperties()
+                    .Where(x => x.CanWrite)
```

Comment:
do we care about aot scenarios?

Reply:
We don't currently in ML.NET as we don't currently support it because of other issues. If it would be an easy fix to change this code to work with that it wouldn't be bad, but otherwise I don't think its critical.